src/library/impl/meta_reader/database.h

```diff
@@ -95,24 +95,141 @@ namespace xlang::meta::reader
             return true;
         }
 
-        explicit database(std::string_view const& path, cache const* cache = nullptr) : file_view{ path }, m_path{ path }, m_cache{ cache }
+        explicit database(std::vector<uint8_t>&& buffer, cache const* cache = nullptr) : m_buffer{ std::move(buffer) }, m_view{ m_buffer.data(), m_buffer.data() + m_buffer.size() }, m_cache{ cache }
```

Comment:
This diff isn't rendering well. All that really happened here is:
- Moved the code in the current constructor into a private `initialize` method
- Added a constructor that takes `std::vector<uint8_t>`